src/rgw/driver/rados/rgw_lc_tier.cc

```diff
@@ -360,6 +424,30 @@ static bool is_already_tiered(const DoutPrefixProvider *dpp,
   return 0;
 }
 
+bool is_restore_in_progress(const DoutPrefixProvider *dpp,
+                            std::map<std::string, std::string>& headers) {
+  map<string, string> attrs = headers;
+
+  for (const auto& a : attrs) {
+    ldpp_dout(dpp, 20) << "GetCrf attr[" << a.first << "] = " << a.second <<dendl;
+  }
+  string s = attrs["X_AMZ_RESTORE"];
+
+  if (s.empty())
+    s = attrs["x_amz_restore"];
+
+  ldpp_dout(dpp, 0) << "is_already_tiered attrs[X_AMZ_RESTORE] = " << s <<dendl;
+
+  if (!s.empty()){
+    const char *r_str = "ongoing-request=\"true\"";
+    char *found = strstr((char*)s.c_str(), r_str);
```

Comment:
nit -- why is there a cast to `(char*)`? `strstr` takes a `const char*` as a parameter and that's what `c_str()` returns.

Reply:
Done.